src/rpc/multipass.proto

```diff
@@ -204,26 +206,66 @@ message InstanceStatus {
     Status status = 1;
 }
 
+message InstanceSpecificDetails {
+    string image_release = 1;
+    string current_release = 2;
+    string id = 3;
+    string load = 4;
+    string memory_usage = 5;
+    string disk_usage = 6;
+    repeated string ipv4 = 7;
+    repeated string ipv6 = 8;
+    int32 num_snapshots = 9;
+}
+
+message SnapshotFundamentalInfo {
+    string snapshot_name = 1;
+    string parent = 2;
+    string comment = 3;
+}
+
+message SnapshotSpecificDetails {
+    SnapshotFundamentalInfo fundamentals = 1;
+    string size = 2;
+    int32 creation_timestamp = 3; // TODO@snapshots use timestamp type
+    repeated string children = 4;
+}
+
+message DetailedInfoItem {
+    string instance_name = 1;
+    InstanceStatus instance_status = 2;
+    string memory_total = 3;
+    string disk_total = 4;
+    string cpu_count = 5;
+    MountInfo mount_info = 6;
+
+    oneof extra_info {
+        InstanceSpecificDetails instance_info = 7;
+        SnapshotSpecificDetails snapshot_info = 8;
+    }
+}
+
+message SnapshotOverviewInfoItem {
+    string instance_name = 1;
+    SnapshotFundamentalInfo fundamentals = 2;
```

Comment:
Need an additional field here to store a timestamp so that snapshots can be sorted. I think you should be able to use an `int64` value which is smaller than a `Timestamp` and then populate it with the unix epoch time.

Reply:
Ah, good point. Need to pull it from the details.

Reply:
But if we can make `Timestamp` work, I think that would be better.